sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFn.java

```diff
@@ -758,11 +804,19 @@ public Duration getAllowedTimestampSkew() {
    * Annotation for the method that maps an element to an initial restriction for a <a
    * href="https://s.apache.org/splittable-do-fn">splittable</a> {@link DoFn}.
    *
-   * <p>Signature: {@code RestrictionT getInitialRestriction(InputT element, <optional arguments>);}
+   * <p>Signature: {@code RestrictionT getInitialRestriction(<arguments>);}
    *
-   * <p>The optional arguments are allowed to be:
+   * <p>This method must satisfy the following constraints:
    *
    * <ul>
+   *   <li>The return type {@code RestrictionT} defines the restriction type used within this
+   *       splittable DoFn. All other methods that use a {@link Restriction @Restriction} parameter
+   *       must use the same type that is used here. It is suggested to use as narrow of a return
+   *       type definition as possible (for example prefer to use a square type over a shape type as
+   *       a square is a type of a shape).
+   *   <li>If one of its arguments is tagged with the {@link Element} annotation, then it will be
+   *       passed the current element being processed; the argument must be of type {@code InputT}.
+   *       Note that schema element parameters are currently unsupported.
```

Comment:
I would also suggest updating the DoFn documentation related to schemas since the docs around what is FieldAccess and how @Element parameters interact seems to be incorrect since the documentation currently states that "the argument type must match the input type of this DoFn"

Reply:
good point. I filed [BEAM-9217](https://issues.apache.org/jira/browse/BEAM-9217) to follow-up with that.